sklearn/metrics/_regression.py

```diff
@@ -585,13 +585,21 @@ def root_mean_squared_error(
     >>> y_true = [3, -0.5, 2, 7]
     >>> y_pred = [2.5, 0.0, 2, 8]
     >>> root_mean_squared_error(y_true, y_pred)
-    np.float64(0.612...)
+    0.612...
     >>> y_true = [[0.5, 1],[-1, 1],[7, -6]]
     >>> y_pred = [[0, 2],[-1, 2],[8, -5]]
     >>> root_mean_squared_error(y_true, y_pred)
-    np.float64(0.822...)
+    0.822...
     """
-    output_errors = np.sqrt(
+
+    xp, _ = get_namespace(y_true, y_pred, sample_weight, multioutput)
+    dtype = _find_matching_floating_dtype(y_true, y_pred, xp=xp)
+
+    y_type, y_true, y_pred, multioutput = _check_reg_targets(
+        y_true, y_pred, multioutput, dtype=dtype, xp=xp
+    )
+    check_consistent_length(y_true, y_pred, sample_weight)
```

Comment:
Do we really need these checks here? I think since this calls `mean_squared_error` the checks will be performed within that. As for `root_mean_squared_log_error` which calls this we are also performing the checks there.

Reply:
TL;DR: The `check_consistent_length` part is redundant in both `mean_squared_log_error` and `root_mean_squared_log_error`.

Since the `<= -1` condition must be verified with the Array API:
1. We need to establish the correct namespace variable `xp`, as the inequality applies only to `y_true` and `y_pred`. The correct `xp` is determined by `get_namespace(y_true, y_pred)`, rather than including `sample_weight` and `multioutput`.

2. We also need to convert `y_true` and `y_pred` into array-like objects when verifying the inequality, which is handled by the `_check_reg_targets function`.

These two steps ensure that the `xp.any(...)` parts work correctly.

As mentioned, the `check_consistent_length` part is already performed within `MSE` and `RMSE`. Therefore, we can rely on them to verify it and focus on checking the inequality within `MSLE` and `RMSLE`.

I've updated both `mean_squared_log_error` and `root_mean_squared_log_error` accordingly. It's ready for review. Please let me know if there’s anything you’d like me to change.

Thank you for your time!

Reply:
Thanks!